cellarium/ml/transforms/transforms.py

```diff
@@ -1,108 +1,251 @@
 # Copyright Contributors to the Cellarium project.
 # SPDX-License-Identifier: BSD-3-Clause
 
+from functools import cache
+from typing import Any
+
+import numpy as np
 import torch
+from numpy.typing import ArrayLike
 from torch import nn
 
 
-class NormalizeTotal(nn.Module):
+class DivideByScale(nn.Module):
     """
-    Normalize total gene counts per cell to target count.
+    Divide gene counts by a scale.
+
+    .. math::
+
+        y_{ng} = \\frac{x_{ng}}{\\mathrm{scale}_g + \\mathrm{eps}}
 
     Args:
-        target_count:
-            Target gene epxression count.
+        scale_g:
+            A scale for each gene.
         eps:
             A value added to the denominator for numerical stability.
+        feature_schema:
+            The variable names schema for the input data validation.
+            If ``None``, no validation is performed.
     """
 
-    def __init__(
-        self,
-        target_count: int = 10_000,
-        eps: float = 1e-6,
-    ):
+    def __init__(self, scale_g: torch.Tensor, eps: float = 1e-6, feature_schema: ArrayLike | None = None) -> None:
```

Comment:
Let's not allow the case where feature_schema is None. Instead, let's use caching to make validations fast (see below).

Reply:
I will do the same for `ZScore` then?